scripts/vis_poses.py

```diff
@@ -296,49 +305,197 @@ def main(args):
+                    depth=depth,
+                    vis_rgb_resolve_visib=args.vis_rgb_resolve_visib,
+                    vis_rgb=False,
+                    vis_depth_diff=False,
+                )["res_per_obj"]
+                bres_gt = get_depth_map_and_obj_masks_from_renderings(res_per_obj_gt)
+                mask_objs_gt = bres_gt["mask_objs"]
+                bres_est = get_depth_map_and_obj_masks_from_renderings(res_per_obj_est)
+                mask_objs = bres_est["mask_objs"]
+
+                if "depth_heatmap" in args.extra_vis_types:
+
+                    depth_diff_meshs = []
+                    for obj_idx, obj_res_gt in res_per_obj_gt.items():
+                        obj_res = res_per_obj_est[obj_idx]
+                        gt_depth = obj_res_gt["depth"]
+                        obj_id = obj_res_gt["pose"]["obj_id"]
+                        sym_mat = misc.get_symmetry_transformations(
+                            models_info[obj_id], 0.01
+                        )
+                        pose = obj_res["pose"]
+                        depth_diff_mesh = get_depth_diff_img(
+                            gt_depth=gt_depth,
+                            est_pose=pose,
+                            gt_pose=obj_res_gt["pose"],
+                            cam=cam,
+                            syms=sym_mat,
+                        )
+                        depth_diff_meshs.append(depth_diff_mesh)
+
+                    dres = combine_depth_diffs(
+                        mask_objs_gt,
+                        depth_diff_meshs,
+                    )
+                    depth_diff_img = dres["combined"]
+                    mask_objs_gt_merged = merge_masks(mask_objs_gt)
+                    plot_depth(
+                        depth_diff_img,
+                        cbar_title="Error (mm)",
+                        ax=None,
+                        cmap="turbo",
+                        use_horiz_cbar=True,
+                        use_white_bg=True,
+                        include_colorbar=True,
+                        use_fixed_cbar=True,
+                        mask=mask_objs_gt_merged,
+                    )
+                    save_path = vis_path_base(suffix="_depth_heatmap")
+                    plt.savefig(save_path, dpi=100, bbox_inches='tight')
+                if "contour" in args.extra_vis_types:
+                    contour_img = copy.deepcopy(rgb)
+                    for idx in range(len(res_per_obj_est)):
+                        depth_obj_gt = res_per_obj_gt[idx]["depth"]
+                        depth_obj = res_per_obj_est[idx]["depth"]
+                        depth_obj_mask = depth_obj_gt > 0
+                        mask_obj = mask_objs[idx]
+                        mask_obj_gt = mask_objs_gt[idx]
+
+                        percent = calc_mask_visib_percent(mask_obj, depth_obj_mask)
+                        if percent < 20:
+                            print(f"{idx=} {percent=}")
```

Comment:
use misc.log, where is this 20% coming from? Could be put as a GLOBAL_VARIABLE or arg and explained